### ISSUE TITLE
V3.0 End of Words At End of Sentences Cut Off

### ISSUE THREAD

**Issue by FJCCOMMISH** [JarodMica/audiobook_maker#69]
Other than adding an extra word, then editing the audio, is there a way to avoid cutting off the last sounds of the last word in a sentence?

For example this sentence: "This is the book for you if you want to explore the history of democracy and its shaping forces."

It is read output as this: "This is the book for you if you want to explore the history of democracy and its shaping force." The ending s sound is missing.

This happens often, with many sentences.

**Comment by JarodMica** [comment]
I believe this isn't an issue with the processing of the file, but an unfortunate issue with tortoise TTS. It can be mitigated through training, but I'm not sure of your comfortablilty of training tortoise TTS models. I'm almost done with the styletts API and it's much more accurate at pieces like that, and then even more so, should be adding F5TTS soon as well so look forward to both of those additions!

**Comment by edbartz** [comment]
In v2 of the audiobook maker, I saw some improvement with this issue in tortoise by adding a space before the period at the end of sentences.  I hope this helps.

**Comment by FJCCOMMISH** [comment]
> In v2 of the audiobook maker, I saw some improvement with this issue in tortoise by adding a space before the period at the end of sentences. I hope this helps.

Yes, that did address the dropped last sound, but not at the end of the final sentence. It voices that one, but cuts it off.

**Comment by FJCCOMMISH** [comment]
> I believe this isn't an issue with the processing of the file, but an unfortunate issue with tortoise TTS. It can be mitigated through training, but I'm not sure of your comfortablilty of training tortoise TTS models. I'm almost done with the styletts API and it's much more accurate at pieces like that, and then even more so, should be adding F5TTS soon as well so look forward to both of those additions!

I train with Tortoise and RVC. What can be done in training to address the cutoff?

I just tried styletts API. Maybe it's my samples, but every time it creates static instead of the final word - odd.

**Comment by JarodMica** [comment]
Going to close as this is a model question and not necessarily AB - cutoff may be resolved in training by training with longer audio samples.
